src/basic/fd-util.h

```diff
@@ -104,6 +104,7 @@ static inline int make_null_stdio(void) {
 
 int fd_reopen(int fd, int flags);
 int fd_reopen_condition(int fd, int flags, int mask, int *ret_new_fd);
+int fd_is_opath(int fd);
```

Comment:
can you switch fsync_directory_of_file() over to this? it does a similar check already, manually

Reply:
To cut things short I just prepped #26462.